### PR TITLE
Added missing save after iam mapping is created

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes https://github.com/weaveworks/eksctl/issues/4621

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by aclevername — APPROVED** [review]

File: pkg/actions/karpenter/create.go [hunk 1]
```diff
@@ -69,6 +69,9 @@ func (i *Installer) Create() error {
 	if err := acm.AddIdentity(id); err != nil {
 		return fmt.Errorf("failed to add new identity: %w", err)
 	}
+	if err := acm.Save(); err != nil {
+		return fmt.Errorf("failed to save the identity config: %w", err)
+	}
```
Comment (aclevername) [inline pkg/actions/karpenter/create.go:74]: unit test?

Comment (Skarlso) [reply pkg/actions/karpenter/create.go:74]: Not testing that part. It's not mocked. I'm trying to write something though.

**Review by cPu1 — APPROVED** [review]

**Comment by darraghenright** [comment]
Hi folks 👋 

Great to see this was resolved so quickly! Thanks @Skarlso 😎 

Was this part of the v0.80.0 release?

**Comment by Skarlso** [comment]
@darraghenright Hi! Yes, indeed it was. :)

**Pull request merged**